File: R_shallow/temp.R
rm(list = ls())
source("Main.R")
source("TargetFunctions.R")
library(rlist)

switch <- 'fixed_iter'
of <- rosenbrock

for(dim in c(3,5,10,20,50,70)){
    res_list <- list()
    for(alg in c('de', 'cmaes', 'des')){
        if(switch == 'fixed_iter'){
            res <- optimize(objective_function=of, problem_dimension=dim, , algorithm=alg, max_iter=400)
            res_list <- list.append(res_list, res)
        }else if (switch == 'fixed_tol') {
            res <- optimize(objective_function=of, problem_dimension=dim, , algorithm=alg, tolerance=1e-6)
            res_list <- list.append(res_list, res)
        } 
    }
    # print(res_list)
    str_2_print <- toString(dim)
    for(r in res_list){
        str_2_print <- paste(str_2_print,' & $\num{', sep="")

        if(switch=='fixed_iter'){
            fit <- r@best_point['fitness']
            str_2_print <- paste(str_2_print, toString(fit), sep="")

        }else if(switch == 'fixed_tol'){
            iter <- r@iterations
            str_2_print <- paste(str_2_print, toString(iter), sep="")
        }

        str_2_print <- paste(str_2_print,'}$', sep="")
    }
    str_2_print <- paste(str_2_print,'\\', sep="")
    print(str_2_print)
}

File: Python_shallow/temp.py
from Main import optimize
import TargetFunctions

switch = 'fixed_iter'

out_dict = {}
c = None
for dim in [3, 5, 10, 20, 50, 70]:
    # for dim in range(1, 50, 5):
    res_dict = {}
    for alg in ['de', 'cmaes', 'des']:
        if switch == 'fixed_iter':
            res = optimize(objective_function=TargetFunctions.rastrigin,
                           problem_dimension=dim, plot_data=False, parameter_dict=None,
                           algorithm=alg, time_eval=False, max_iter=400, constraints=c, constraint_handle='projection')
            res_dict[alg] = "{: 0.3e}".format(res.best_point.value)

        if switch == 'fixed_tol':
            res = optimize(objective_function=TargetFunctions.sphere,
                           problem_dimension=dim, plot_data=False, parameter_dict=None,
                           algorithm=alg, time_eval=False, tolerance=1e-5, max_iter=1e4, constraints=c, constraint_handle='projection')
            res_dict[alg] = res.iteration_num

    out_dict[dim] = res_dict
    str_2_print = f'{dim}'
    for num in res_dict.values():
        str_2_print += f' & $\\num{{{num}}}$'
    str_2_print += ' \\\\'
    print(str_2_print)

print(out_dict)


# print(dim, res.best_point.value)
